### PR TITLE
Fix: Fix steep not working on windows

### PR DESCRIPTION
**Pull request by sevenc-nanashi**
This PR will fix steep not working on windows.
Note the test for windows is not done yet. (It deadlocks somehow)

### CODE REVIEW

File: .github/workflows/ruby-windows.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Run test on Windows
```
Comment (soutaro) [inline .github/workflows/ruby-windows.yml:1]: 👏 👏 👏

File: lib/steep/project.rb [hunk 2]
```diff
@@ -16,22 +16,10 @@ def base_dir
...
-               end
-               Pathname.new(
-                 path_str
-               )
-             else
-               orig_path
-             end
+    def relative_path(path)
       path.relative_path_from(base_dir)
+    rescue ArgumentError
```
Comment (soutaro) [inline lib/steep/project.rb:21]: Could you tell me what is raising `ArgumentError` here?

Comment (sevenc-nanashi) [reply lib/steep/project.rb:21]: Pathname#relative_path_from raises ArgumentError when Drive letter is different.

Comment (soutaro) [reply lib/steep/project.rb:21]: Got it. It's really interesting.
Let's think about it in the future, when problem happens.

**Pull request merged**